mlem/contrib/fastapi.py

```diff
@@ -6,15 +6,18 @@
 import logging
 from collections.abc import Callable
 from types import ModuleType
-from typing import Any, ClassVar, List, Type
+from typing import Any, ClassVar, Dict, List, Optional, Tuple, Type
```

Comment:
When I run your [Torchvision example with resnet](https://github.com/iterative/mlem/pull/447#issuecomment-1299909225), I have two handlers: `__call__` and `predict`. While the second works, the first always throws errors like:

<details>

```shell
INFO:     127.0.0.1:60666 - "POST /__call__ HTTP/1.1" 500 Internal Server Error
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/Users/aguschin/.local/share/virtualenvs/mlem-Utz6DvOn/lib/python3.10/site-packages/uvicorn/protocols/http/h11_impl.py", line 404, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
  File "/Users/aguschin/.local/share/virtualenvs/mlem-Utz6DvOn/lib/python3.10/site-packages/uvicorn/middleware/proxy_headers.py", line 78, in __call__
    return await self.app(scope, receive, send)
  File "/Users/aguschin/.local/share/virtualenvs/mlem-Utz6DvOn/lib/python3.10/site-packages/fastapi/applications.py", line 270, in __call__
    await super().__call__(scope, receive, send)
  File "/Users/aguschin/.local/share/virtualenvs/mlem-Utz6DvOn/lib/python3.10/site-packages/starlette/applications.py", line 124, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/Users/aguschin/.local/share/virtualenvs/mlem-Utz6DvOn/lib/python3.10/site-packages/starlette/middleware/errors.py", line 184, in __call__
    raise exc
  File "/Users/aguschin/.local/share/virtualenvs/mlem-Utz6DvOn/lib/python3.10/site-packages/starlette/middleware/errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "/Users/aguschin/.local/share/virtualenvs/mlem-Utz6DvOn/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 75, in __call__
    raise exc
  File "/Users/aguschin/.local/share/virtualenvs/mlem-Utz6DvOn/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 64, in __call__
    await self.app(scope, receive, sender)
  File "/Users/aguschin/.local/share/virtualenvs/mlem-Utz6DvOn/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 21, in __call__
    raise e
  File "/Users/aguschin/.local/share/virtualenvs/mlem-Utz6DvOn/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 18, in __call__
    await self.app(scope, receive, send)
  File "/Users/aguschin/.local/share/virtualenvs/mlem-Utz6DvOn/lib/python3.10/site-packages/starlette/routing.py", line 680, in __call__
    await route.handle(scope, receive, send)
  File "/Users/aguschin/.local/share/virtualenvs/mlem-Utz6DvOn/lib/python3.10/site-packages/starlette/routing.py", line 275, in handle
    await self.app(scope, receive, send)
  File "/Users/aguschin/.local/share/virtualenvs/mlem-Utz6DvOn/lib/python3.10/site-packages/starlette/routing.py", line 65, in app
    response = await func(request)
  File "/Users/aguschin/.local/share/virtualenvs/mlem-Utz6DvOn/lib/python3.10/site-packages/fastapi/routing.py", line 231, in app
    raw_response = await run_endpoint_function(
  File "/Users/aguschin/.local/share/virtualenvs/mlem-Utz6DvOn/lib/python3.10/site-packages/fastapi/routing.py", line 162, in run_endpoint_function
    return await run_in_threadpool(dependant.call, **values)
  File "/Users/aguschin/.local/share/virtualenvs/mlem-Utz6DvOn/lib/python3.10/site-packages/starlette/concurrency.py", line 41, in run_in_threadpool
    return await anyio.to_thread.run_sync(func, *args)
  File "/Users/aguschin/.local/share/virtualenvs/mlem-Utz6DvOn/lib/python3.10/site-packages/anyio/to_thread.py", line 31, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
  File "/Users/aguschin/.local/share/virtualenvs/mlem-Utz6DvOn/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 937, in run_sync_in_worker_thread
    return await future
  File "/Users/aguschin/.local/share/virtualenvs/mlem-Utz6DvOn/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 867, in run
    result = context.run(func, *args)
  File "/Users/aguschin/Git/iterative/mlem/mlem/contrib/fastapi.py", line 139, in handler
    result = executor(**{arg_name: arg})
  File "/Users/aguschin/Git/iterative/mlem/mlem/runtime/interface.py", line 204, in executor
    return self.model_type.call_method(method_name, *args)
  File "/Users/aguschin/Git/iterative/mlem/mlem/core/model.py", line 278, in call_method
    output_data = self._call_method(signature.name, *input_data)
  File "/Users/aguschin/Git/iterative/mlem/mlem/core/model.py", line 293, in _call_method
    return getattr(self.model, wrapped)(*input_data)
AttributeError: 'function' object has no attribute 'model_f'
```

</details>

PS: I see that there are images that `predict` cannot handle, but the errors are different (like `RuntimeError: The size of tensor a (4) must match the size of tensor b (3) at non-singleton dimension 0`). So this is something wrong with the handler.

BTW, can it be fixed by https://github.com/iterative/mlem/pull/456 by chance?

Reply:
Looks like different issue, can you try it without torch?

Reply:
yes, https://github.com/iterative/mlem/issues/475